Yosemite/Yosemite/Stores/MetaDataStore.swift

```diff
@@ -117,9 +117,18 @@ private extension MetaDataStore {
                                                siteID: Int64,
                                                onCompletion: @escaping () -> Void) {
         let derivedStorage = sharedDerivedStorage
+
         derivedStorage.perform {
+            let storedMetaData = derivedStorage.loadOrderMetaData(siteID: siteID, orderID: orderID)
+
             for readOnlyOrderMetaData in readOnlyOrderMetaDatas {
-                self.saveMetaData(derivedStorage, readOnlyOrderMetaData, orderID: orderID, siteID: siteID)
+                self.saveMetaData(derivedStorage, readOnlyOrderMetaData, storedMetaData, orderID: orderID, siteID: siteID)
```

Comment:
I'm passign the `storedMetaData` to the `saveMetaData` function to avoid doing two calls to fetch the items from storage, please let me know if this is not needed.